### PR TITLE
Transaction frontend

### PR DESCRIPTION
**Pull request by Andrius-B**
### Added a _dumb_ transaction form
_Change index component `<App />` to `<TransactionForm />` if you want to see it_

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -9,8 +9,10 @@
     "@types/node": "^11.9.4",
     "@types/react": "^16.8.3",
     "@types/react-dom": "^16.8.2",
+    "@types/react-router-dom": "^4.3.1",
     "react": "^16.8.2",
     "react-dom": "^16.8.2",
+    "react-router-dom": "^4.3.1",
```
Comment (Martynas-S) [inline frontend/package.json]: Butu gerai, kad vietoj situ ir package-lock.json pakeitimu tiesiog imergintum development i sita brancha

File: frontend/src/Components/TransactionForm/TagPicker.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        );
+    };
+
+    if (disabled && tagError) {
+        setTagError(false);
+    }
+    return(
+        <React.Fragment>
+            <Grid item={true} xs={12} sm={12} className={classes.chips}>
+                {renderTags()}
```
Comment (Martynas-S) [inline frontend/src/Components/TransactionForm/TagPicker.tsx]: Kodel ne tiesiog `<ChipArray chips={tags} onRemove={removeTag}/>` ?? :D

File: frontend/src/Components/TransactionForm/TransactionForm.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  const onFieldChange = (field: string, value: any) => {
+    const clone = {...transaction} as any;
+    clone[field] = value;
+    setTransaction(clone);
+  };
+
+  const onSubmit = () => {
+    // console.log(transaction);
+    setLoading(true);
+    const _ = (props.onSubmit && props.onSubmit(transaction));
```
Comment (Martynas-S) [inline frontend/src/Components/TransactionForm/TransactionForm.tsx]: Geriau pakeisk sita i normalu if statement :D

**Review by Martynas-S — CHANGES_REQUESTED** [review]
Dar butu gerai kad mes naudotume unified file hierachy structure, a la:
├── Components
│   └── FormField.tsx
└── Views
    ├── Login
    │   ├── LoginForm.tsx
    │   └── Login.tsx
    └── Register
        ├── Register.tsx
        └── RegistrationForm.tsx

**Comment by Martynas-S** [comment]
Ai ir dar butu gerai imergint development del neveikianciu testu xd

**Comment by Martynas-S** [comment]
Transaction forma leidzia irasyti neigiama price

**Comment by Martynas-S** [comment]
Reikia pakurti preliminaru route pries merginant i development

**Comment by Martynas-S** [comment]
Transactionas gali buti expense arba income

**Review by Martynas-S — APPROVED** [review]

File: frontend/src/serviceWorker.ts [hunk 4]
```diff
@@ -121,13 +130,13 @@ function checkValidServiceWorker(swUrl, config) {
     })
     .catch(() => {
       console.log(
-        'No internet connection found. App is running in offline mode.',
+        "No internet connection found. App is running in offline mode.",
```
Comment (erteriuke) [inline frontend/src/serviceWorker.ts:133]: Great practice, thanks for looking ahead! <3

**Review by erteriuke — APPROVED** [review]

**Pull request merged**